cpp/src/io/json/json_tree.cu

```diff
@@ -28,15 +28,15 @@
 #include <cudf/utilities/error.hpp>
 #include <cudf/utilities/span.hpp>
 
-#include <cuco/static_map.cuh>
-
-#include <cub/device/device_radix_sort.cuh>
-
 #include <rmm/cuda_stream_view.hpp>
 #include <rmm/device_uvector.hpp>
 #include <rmm/exec_policy.hpp>
 #include <rmm/mr/device/polymorphic_allocator.hpp>
 
+#include <cub/device/device_radix_sort.cuh>
+
+#include <cuco/static_map.cuh>
```

Comment:
I believe that `cub` and `thurst` headers should stay closer together than with `cuco`.

Reply:
My understanding of the include order:

1. local dir
2. cuIO
3. libcudf
4. rmm
5. other
6. std C++ lib

Is there a whole hierarchy in "other" that I'm missing? :D

Reply:
I meant, the order should be thrust=>cub (or cub=>thrust) => cuco, instead of cub => cuco => thrust like this.

Reply:
could you please explain why?
I assumed these are all equal so I sorted alphabetically.

Reply:
Because `thrust` internally calls `cub` so they are more related so I think they should stay together (in theory).
In reality that doesn't matter at all :smile:

Reply:
AFAIK the ordering in general aims to be such that the lowest dependencies are listed last, so we don't cover up missing dependencies by accident.
I don't think we need to be as cautious with the order as you suggest, but I'll make the change if you think it's warranted.

Reply:
That makes sense to me. It doesn't matter much---very minor nit.